### ISSUE TITLE
future.globals does not seem to work when variables are "created" in the named list

### ISSUE THREAD

**Issue by DavisVaughan** [HenrikBengtsson/future.apply#10]
Kind of an obscure issue but definitely an issue:

``` r
library(future.apply)

plan(sequential)
future_lapply(
  X = 1, 
  FUN = function(x) {y}, 
  future.globals = list(y = 4)
)
#> Error in ...future.FUN(...future.X_jj, ...): object 'y' not found

z <- 1
future_lapply(
  X = 1, 
  FUN = function(x) {y}, 
  future.globals = list(y = z)
)
#> Error in ...future.FUN(...future.X_jj, ...): object 'y' not found
```

Created on 2018-05-09 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

Interestingly, the `y` value is recognized as a global, and makes it into the environment that the expression is evaluated in. You can see this by debugging and making your way to [this](https://github.com/HenrikBengtsson/future/blob/1e9d3f832137650b4ac2743b60937cc8adba8e27/R/UniprocessFuture-class.R#L61) line of `run.UniprocessFuture()`. Not sure why it can't be seen inside the lapply() after that

**Comment by DavisVaughan** [comment]
Seems to not be a problem with pure futures

``` r
library(future.apply)
#> Loading required package: future
#> 
#> Attaching package: 'future.apply'
#> The following object is masked from 'package:future':
#> 
#>     future_lapply
plan(sequential)
value(future({y}, globals = list(y = 4)))
#> [1] 4

value(future(
  expr = { lapply(1, function(x) {y}) }, 
  globals = list(y = 4)
))
#> [[1]]
#> [1] 4
```

Created on 2018-05-09 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by HenrikBengtsson** [comment]
Thxs. Here's an example of the problem using plan futures:
```r
> library(future)
> plan(sequential)
> f <- futureCall(function(x) y, args = list(x = 1), future.globals = list(y = 3))
> value(f)
Error in (function (x)  : object 'y' not found
```
It'll be solved over at https://github.com/HenrikBengtsson/future/issues/225.  Closing this one.

**Comment by HenrikBengtsson** [comment]
My `futureCall()` example was invalid.  Re-opening this future.apply issue for now (though the solution to it will still be in the future package).

**Comment by HenrikBengtsson** [comment]
Turns out that I just fixed this in **future** (>= 1.21.0-9003) in https://github.com/HenrikBengtsson/future/commit/6a87881cbe6d91be544a212e9470d8eabed630f5, cf. https://github.com/HenrikBengtsson/future/issues/515.
